tests/shardddl3/run.sh

```diff
@@ -871,6 +871,1199 @@ function DM_112() {
+
+    run_sql_source2 "alter table ${shardddl1}.${tb1} drop index idx, add index idx(c3,c1,c2);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(7,7,7,7);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(8,8,8,8);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(9,9,9,9);"
+
+    run_sql_source2 "alter table ${shardddl1}.${tb2} drop index idx, add index idx(c3,c1,c2);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(10,10,10,10);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(11,11,11,11);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(12,12,12,12);"
+
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+}
+
+
+# Adjust index fields combination.
+function DM_118 {
+    run_case 118 "double-source-pessimistic" \
+    "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int primary key, c1 int, c2 int, c3 int, index idx(c1, c2, c3));\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb1} (a int primary key, c1 int, c2 int, c3 int, index idx(c1, c2, c3));\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb2} (a int primary key, c1 int, c2 int, c3 int, index idx(c1, c2, c3));\"" \
+    "clean_table" "pessimistic"
+    run_case 118 "double-source-optimistic" \
+    "run_sql_source1 \"create table ${shardddl1}.${tb1} (a int primary key, c1 int, c2 int, c3 int, index idx(c1, c2, c3));\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb1} (a int primary key, c1 int, c2 int, c3 int, index idx(c1, c2, c3));\"; \
+     run_sql_source2 \"create table ${shardddl1}.${tb2} (a int primary key, c1 int, c2 int, c3 int, index idx(c1, c2, c3));\"" \
+    "clean_table" "optimistic"
+}
+
+function DM_119_CASE {
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(1,1,1,1);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(2,2,2,2);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(3,3,3,3);"
+
+    run_sql_source1 "alter table ${shardddl1}.${tb1} add index idx(c1,c2);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(4,4,4,4);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(5,5,5,5);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(6,6,6,6);"
+
+    run_sql_source2 "alter table ${shardddl1}.${tb1} add index idx(c1,c3);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} values(7,7,7,7);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(8,8,8,8);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(9,9,9,9);"
+    
+    # FIXME: DM should detect conflicts and give human readable error messages.
+    # For example:
+    # if [[ "$1" = "pessimistic" ]]; then
+    #     check_log_contain_with_retry "is different with" $WORK_DIR/master/log/dm-master.log
+    # else
+    #     run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+    #         "query-status test" \
+    #         "because schema conflict detected" 1
+    # fi
```

Comment:
Can pessimistic detect it? We may also add the negative check now. e.g.
```
check_sync_diff $WORK_DIR $cur/conf/diff_config.toml 10 "fail"
```

Reply:
Yes, `check_sync_diff`  can detect in both pessimistic and optimistic mode. But clear error messages and logs are still needed.